### ISSUE TITLE
Log file is locked until the execution of code finished

### ISSUE THREAD

**Issue by littleQiu22** [scipopt/PySCIPOpt#587]
First thanks for the effort of PySCIPOpt team and community to offer such neat and elegant API :)
I use SCIP version 8.0.0 and have a problem about locked log file . 

**The phenomenon**:
- When set log file for `Model`, then the log file is locked and it cannot be deleted or moved until the execution of code finished. 

**The question**:
- Can we free the lock of log file when execution of code goes? For example,  use a function like `model.freeLogFileLock()` to acheieve this goal. (I guess this problem is not important, so a brief reply can be enough for me, thanks in advance)

**The minimum code to repeat the case**
- Example of locked file  
`from pyscipopt import Model`
`import os`
`model = Model()`
`x=model.addVar('x')`
`model.setObjective(x,'minimize')`
`model.setLogfile('<My file name>')`
`model.optimize()`
`os.remove('<My file name>') # delete locked file when execution of code goes, and then error occurs`
Then python will complain that "The file is occupied by another process".

**Comment by CGraczyk** [comment]
Hi, thanks for using the PySCIPOpt API - my first question here would be why this would be necessary and how this would be used since I'd assume this could result in errors on the SCIP side. So my first intuition would be no - i dont think this is easily done without deeper changes in the underlying SCIP code.

**Comment by littleQiu22** [comment]
@CGraczyk First thanks for your response. Next  I will answer your questions briefly and close my issue.

- Necessity: Actually the feature proposed in my question is **unnecessary** and **unimportant**.
- Usage: 

1. The phenomenon of locked log-file can make operation on log-file in **interactive shell** (e.g. Jupyter Notebook) fail. 
2. If I set `model.setLogfile('<file name>')` and optimize in Jupyter Notebook, then the log file is locked by SCIP until the Notebook kernel is restarted or `model` is deleted. 
3. Above scenario is what I experienced. So I think if there are more elegant solutions than deleting `model` or restarting notebook kernel, the **interactive shell** can be used more smoothly.

- Conclusion: Thanks for your response, my question is answered.
